tests/jquery.flot.composeImages.Test.js

```diff
@@ -0,0 +1,674 @@
+/* eslint-disable */
+/* global $, describe, it, xit, xdescribe, after, afterEach, expect*/
+
+describe("composeImages", function() {
+    var placeholder, plot;
+    var composeImages = $.plot.composeImages;
+
+
+    function matchPixelColor(pixelData, r, g, b, a) {
+        return (pixelData[0] === r) && (pixelData[1] === g) && (pixelData[2] === b) && (pixelData[3] === a);
+    }
+
+
+    function matchPixelColorWithError(pixelData, r, g, b, a, err) {
+        return (Math.abs(pixelData[0] - r) <= err) && (Math.abs(pixelData[1] - g) <= err) && (Math.abs(pixelData[2] - b) <= err) && (Math.abs(pixelData[3] - a) <= err);
+    }
+
+    function matchPixelDataArrays(pixelData1, pixelData2) {
+        var sameValue = true;
+        if (pixelData1.length !== pixelData2.length) {
+            sameValue = false;
+        } else {
+            for (var i = 0; i < pixelData1.length; i++) {
+                if (pixelData1[i] !== pixelData2[i]) {
+                    sameValue = false;
+                    break;
+                }
+            }
+        }
+        return sameValue;
+    }
+
+    beforeEach(function() {
+        placeholder = setFixtures('<div id="test-container" style="width: 600px;height: 400px; padding: 0px margin: 0px; border: 0px; font-size:0pt; font-family:sans-serif; line-height:0px;">')
+            .find('#test-container');
+    });
+
+    it('should call composeImages on an empty array of sources, so the destination canvas should stay unmodified', function (done) {
+        var sources = placeholder.html('<div id="test-container" style="width: 600px;height: 400px">' +
+        '</div>' +
+        '<canvas id="myCanvas" width="300" height="150" style="border:1px solid #d3d3d3;"></canvas>'
+        ).find('svg').toArray();
+
+        var destinationCanvas = document.getElementById("myCanvas");
+        var pixelData; //used later
+
+        function writeSomethingToCanvas(canvas) {
+            var ctx = canvas.getContext('2d');
+            //ctx.beginPath();
+            ctx.arc(80, 10, 5, 0, 2 * Math.PI);
+            ctx.fill();
+            //ctx.stroke();
+        }
+
+        writeSomethingToCanvas(destinationCanvas); //make sure composeImages won't modify this content
+
+        composeImages(sources, destinationCanvas).then(function() {
+            pixelData = destinationCanvas.getContext('2d').getImageData(80, 10, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 0, 255)).toBe(true);
+
+            expect(destinationCanvas.width).toBe(300);
+            expect(destinationCanvas.height).toBe(150);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(10, 10, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 0, 0)).toBe(true);
```

Comment:
Uncle Bob is recommending 3 lines of code long tests :-) Let's try to get as close as possible to this recommendation.
You could collapse each pair of these 2 lines into a single line using the Jasmine matchers infrastructure. Here is an example: https://github.com/ni-kismet/flot-intensitygraph-plugin/blob/master/tests/utils/colorMatchers.js#L4

You could define at the bottom of this file your own Jasmine matcher. The best part the custom matcher is that you can better format the error message so a failed expect() will provide a better message than "expected false to be true".